heatmap/heatmap.js

```diff
@@ -33,19 +33,22 @@ var heatmapApp =
 				headInjector.setWindowName();
 				headerInfo.pid = MathUtils.uuid();
 				headerInfo.cid = "heatmap";
-				$rootScope.pid
 				var header = {
                     wid: $window.name,
                     cid: headerInfo.cid,
                     pid: headerInfo.pid,
-					action: "main",
-					schema_table: "Gene_Expression:Array_Data_view"
+					action: "model/read",
+					schema_table: "Gene_Expression:Array_Data_view",
+					catalog: UriUtils.getCatalogId()
 				};
-				ERMrest.resolve(ermrestURI, {cid: headerInfo.cid, pid: headerInfo.pid, wid: $window.name, action: "model/read"}).then(function getReference(reference) {
+				ERMrest.resolve(ermrestURI, header).then(function getReference(reference) {
 					verifyConfiguration(reference);
 					if (!$rootScope.configErrorsPresent) {
 						var sortBy = typeof heatmapConfig.data.sortBy !== "undefined" ? heatmapConfig.data.sortBy : [];
 						var ref = reference.sort(sortBy);
+						header['action']="main";
+						header["ppid"]=UriUtils.getQueryParams(location.href).ppid;
+						header["pcid"]=UriUtils.getQueryParams(location.href).pcid;
```

Comment:
These 2 statements need to be conditionals. We only want to attach ppid and pcid if the values are defined.

Reply:
Made required changes